supportedBackends/jes/src/main/scala/cromwell/backend/impl/jes/labels/Label.scala

```diff
@@ -0,0 +1,59 @@
+package cromwell.backend.impl.jes.labels
+
+/**
+  * Construct via Label.safeLabel
+  */
+case class Label private[labels](key: String, value: String) {
+
+  def +(that: Label): Labels = Labels(Vector(this, that))
+}
+
+object Label {
+
+  def validate(s: String) = "[a-z]([-a-z0-9]*[a-z0-9])?".r.pattern.matcher(s).matches && s.length <= 63
```

Comment:
This is only called from one spot in the production code.  This certainly is an optimization in that it short circuits the hairy "else" at that call site, but overall complexity might be less without this optimization.